homeassistant/components/workday/binary_sensor.py

```diff
@@ -197,28 +199,27 @@ def is_exclude(self, day, now):
 
         return False
 
-    @property
-    def extra_state_attributes(self):
-        """Return the attributes of the entity."""
-        # return self._attributes
-        return {
-            CONF_WORKDAYS: self._workdays,
-            CONF_EXCLUDES: self._excludes,
-            CONF_OFFSET: self._days_offset,
-        }
-
-    async def async_update(self):
+    async def async_update(self) -> None:
         """Get date and look whether it is a holiday."""
         # Default is no workday
-        self._state = False
+        self._attr_is_on = False
 
         # Get ISO day of the week (1 = Monday, 7 = Sunday)
-        date = get_date(dt.now()) + timedelta(days=self._days_offset)
-        day = date.isoweekday() - 1
+        adjusted_date = get_date(dt.now()) + timedelta(days=self._days_offset)
+        day = adjusted_date.isoweekday() - 1
         day_of_week = day_to_string(day)
 
-        if self.is_include(day_of_week, date):
-            self._state = True
+        self._attr_extra_state_attributes = {
+            CONF_WORKDAYS: self._workdays,
+            CONF_EXCLUDES: self._excludes,
+            CONF_OFFSET: self._days_offset,
+        }
+
+        if not day_of_week:
```

Comment:
This change the code behavior:
`day = adjusted_date.isoweekday() - 1` --> so it will range from 0 to 6
`if not day_of_week:` meaning we don't test against 0 in the `is_include` and `is_exclude` bellow.

Would be probably better to change it to:
```suggestion
        if day_of_week is None:
```
However looking at the code I would change the following which will simplify the code:
```python
def day_to_string(day: int) -> str:
    """Convert day index 0 - 7 to string."""
        return ALLOWED_DAYS[day]
```
and:
```python
    try:
        day_of_week = day_to_string(day)
    except IndexError:
        return

Reply:
Did not want to modify the tests so I went with the first fix